### PR TITLE
Add spell check CI and fix typos

### PR DESCRIPTION
**Pull request by ytdHuang**
## Description
As title.

"[typos](https://github.com/crate-ci/typos)" will check our spelling, tell us where each typo is located, and give us suggestions.

I also add the ignore file for the word: `ket`, since we use it a lot.

@albertomercurio 
If you agree with this, I will fix all the typos detected by this CI.

### CODE REVIEW

**Comment by albertomercurio** [comment]
Yes good idea. Do we correct all the spelling in this pr? Or we merge and we fix them in the future ones?

File: src/steadystate.jl [hunk 1]
```diff
@@ -106,11 +106,11 @@ function _steadystate(
     idx_range = collect(1:N)
     rows = _get_dense_similar(L_tmp, N)
     cols = _get_dense_similar(L_tmp, N)
-    datas = _get_dense_similar(L_tmp, N)
+    _data = _get_dense_similar(L_tmp, N)
```
Comment (albertomercurio) [inline src/steadystate.jl:109]: I think it is better to leave `datas` and add it to the words to exclude. We should keep the code readable

Comment (ytdHuang) [reply src/steadystate.jl:109]: but `data` is an uncountable noun @@

Comment (ytdHuang) [reply src/steadystate.jl:109]: or something like `L_tmp2` ?

Comment (ytdHuang) [reply src/steadystate.jl:109]: This is just a temporary variable
So I prefer not adding it to the exclude list

Comment (albertomercurio) [reply src/steadystate.jl:109]: Ok, maybe `nzvals` or `vals`. What I don't like is that is is the only variable written with that underscore.

Comment (ytdHuang) [reply src/steadystate.jl]: ok, changed to `vals`

**Pull request merged**